### PR TITLE
feat: added base material prop to utility shader materials

### PR DESCRIPTION
**Pull request by FarazzShaikh**
### Why

Adds `baseMaterial` prop to `MeshDistortMaterial` and `MeshWobbleMaterial`. This is cool because it lets users choose what type of shading to apply to these materials.

`baseMaterial` can be a constructor to any material from ThreeJS and is an optional prop. Defaults to `MeshStandardMaterial`

```jsx
<MeshWobbleMaterial
  attach="material"
  color="#f25042"
  speed={1}
  factor={1}
  baseMaterial={THREE.MeshPhongMaterial}
/>
```

### What

Used [CustomShaderMaterial](https://github.com/FarazzShaikh/THREE-CustomShaderMaterial) to provide aforementioned feature

### Checklist

- [x] Documentation updated
- [x] Storybook entry added
- [x] Ready to be merged

### CODE REVIEW

File: src/core/MeshWobbleMaterial.tsx
Comment (FarazzShaikh) [reply src/core/MeshWobbleMaterial.tsx:9]: I agree, I'm not a typescript master so I couldn't find a way to do so, any ideas? I know there is `Pick<T, K>` but how to use it in this context is confusing

**Comment by DennisSmolek** [comment]
Question: 
With https://lamina.dev being put together does it make more sense to hold off messing with these until they can be refactored to use Lamina?

Not that this is a bad Idea, but I'd expect more utility materials get added or even just the `useLamina` hooks to make the same effect.

**Comment by FarazzShaikh** [comment]
For that 2 things need to happen with Lamina
- Lamina needs to support Vertex Shaders, doesn't at the moment (its experimental)
- Lamina needs to support standard materials as Layers or a `<Shading  />` layer, it doesn't at the moment 

I can always rewrite these with Lamina when lamina does support said features, I think we should be good to merge this once the Typescript issues are solved

**Comment by FarazzShaikh** [comment]
@joshuaellis Are you working on this PR?

**Review by joshuaellis — CHANGES_REQUESTED** [review]
Question about the deps? And the types need addressing before we can merge this. I'm struggling to find time right not to look at the types, can try again this week.

File: package.json
Comment (FarazzShaikh) [reply package.json:67]: ThreeJS let’s us ether use a barebones shader material or override existing material using onBeforeCompile.

This dependency is just a fancy wrapper around the latter. It will let us inject custom vertex shader into pre-existing materials. Providing for example, Normal, Phong or Physical shading to the WobbleMaterial.

I could remove the dependency but it would require rewriting code that is already written in the dependency 🤷‍♂️

**Comment by FarazzShaikh** [comment]
Sure, I will also give it another go and keep the PR conflict free.

**Comment by FarazzShaikh** [comment]
Closign this one, will make an other PR once CSM's types are more robust. Thanks!

**Pull request closed without merging**
